### DISCUSSION TITLE
Arrays of HashMaps no longer legal in Vyper?

### DISCUSSION THREAD

**Discussion by scherrey** [vyperlang/vyper#2539] [Q&A]
Used to be able to declare an array of HashMaps but this, formerly legal declaration, "levels: public(HashMap(uint256, Bid)[MAX_LEVEL])" now gives this error message: "vyper.exceptions.StructureException: HashMap must be defined with a key type and a value type". How can I get an array of HashMaps with modern Vyper?

**Comment by fubuloubu** [comment]
I don't know if that was ever possible :thinking:

**Reply by scherrey** [reply]
Yep - totally used to compile and worked quite well. Now we need to figure out some way to get the equivalent functionality. Why isn't it legal any more?

**Reply by charles-cooper** [reply]
Which version did it compile on?

**Comment by scherrey** [comment]
Was 0.1.0b12. I haven't had the opportunity to determine when it stopped
working.
